common/environment/environment.cpp

```diff
@@ -1058,6 +1126,75 @@ IConstMachineInfo * CLocalEnvironment::getMachineByAddress(const char * name) co
     return (CConstMachineInfo *) cached;
 }
 
+IConstMachineInfo * CLocalEnvironment::getMachineForLocalHost() const
+{
+    IConstEnvBase *cached = getCache("Hardware/Computer[@netAddress='.']");
+    if (!cached)
```

Comment:
I don't you need most of the implementation here..

if you do:
```
buildMachineCache()
IConstEnvBase *cached = getCache("Hardware/Computer[@netAddress='.']");
```

Then buildMachineCache should have filled the "." cache entry.